release-schema: Add finalStatus to tender, award and contract. Clarify tender.datePublished.

duncandewhurst opened the pull request:
* Closes #1160 (addressing points 3 and 4 from https://github.com/open-contracting/standard/issues/1160#issuecomment-1596302881)

duncandewhurst committed:
release-schema: Add finalStatus to tender, award and contract

duncandewhurst commented:
~~This might need to be merged into #1509 depending on the answer to https://github.com/open-contracting/standard/issues/1160#issuecomment-1758779508~~

duncandewhurst committed:
release-schema: Clarify the description of tender.datePublished

jpmckinney commented:
For #1160 we had started using the phrase "contracting documents (for example, procurement documents)" based on this exchange https://github.com/open-contracting/standard/pull/1509#discussion_r878558410

duncandewhurst committed:
release-schema: Update description of tender.datePublished

duncandewhurst committed:
Update changelog

jpmckinney commented:
Re: 'withdrawn', I had commented in https://github.com/open-contracting/standard/issues/1160#issuecomment-1141320491

> (3. ii.) I'm not sure that we need "no more publication". I'm not aware that anyone has used the 'withdrawn' code for tender/status. At any rate, this is moreso information for the publication policy (e.g. "We only publish up to the initial contract values. We don't publish any modifications."). We have not yet witnessed any demand for "This specific contract will no longer receive updates, unlike other contracts." If we do end up needing it, it deserves its own field, because "no more updates" is orthogonal (i.e. is a separate concern to) the "status" or "final state" of the object. For example, it could be that a contract has reached a final state, but it can still be updated (evaluations of its performance, contract monitoring results, other later assessments, etc.). "No more updates" shouldn't overwrite the "final state" value.

There hasn't been any disagreement yet.

I suggested a change to the deprecation note for tenderStatus' 'withdrawn' in #1509.

duncandewhurst committed:
schema/codelists: Remove withdrawn codes from finalStatus codelists

duncandewhurst commented:
Ah, I'd missed that. Your reasoning sounds good to me so I've removed the 'withdrawn' codes.

duncandewhurst committed:
release-schema: Update enum for finalStatus fields

duncandewhurst committed:
Merge branch '1.2-dev' into 1160-add-finalStatus-fields

odscjen commented on schema/codelists/awardFinalStatus.csv:
```diff
@@ -0,0 +1,4 @@
+Code,Title,Description
+complete,Complete,"The award was made and the standstill period, if any, has ended."
+cancelled,Cancelled,"The award was cancelled by the buyer or the procuring entity (e.g. because of a change in needs, insufficient funds, or technical or procedural errors) after the bid submission deadline, but before the award was made or before the standstill period, if any, had ended."
```
```suggestion
cancelled,Cancelled,"The award was cancelled by the buyer or the procuring entity (for example, because of a change in needs, insufficient funds, or technical or procedural errors) after the bid submission deadline, but before the award was made or before the standstill period, if any, had ended."
```

odscjen commented on schema/codelists/awardFinalStatus.csv:
```diff
@@ -0,0 +1,4 @@
+Code,Title,Description
+complete,Complete,"The award was made and the standstill period, if any, has ended."
+cancelled,Cancelled,"The award was cancelled by the buyer or the procuring entity (e.g. because of a change in needs, insufficient funds, or technical or procedural errors) after the bid submission deadline, but before the award was made or before the standstill period, if any, had ended."
+unsuccessful,Unsuccessful,"The award failed (e.g. a review body cancelled the process during the standstill period) after the bid submission deadline, but before the award was made or before the standstill period, if any, had ended."
```
```suggestion
unsuccessful,Unsuccessful,"The award failed (for example, a review body cancelled the process during the standstill period) after the bid submission deadline, but before the award was made or before the standstill period, if any, had ended."
```

odscjen commented on schema/codelists/contractFinalStatus.csv:
```diff
@@ -0,0 +1,5 @@
+Code,Title,Description
+terminatedSuccessfully,Terminated successfully,"The contract was concluded and in force, and is terminated due to its successful completion."
+terminatedEarly,Terminated early,"The contract was concluded and in force, and is terminated early due to some non-completion."
+cancelled,Cancelled,"The contracting process was cancelled by the buyer or the procuring entity (e.g. because of a change in needs, insufficient funds, or technical or procedural errors) after the award was made and the standstill period, if any, had ended."
```
```suggestion
cancelled,Cancelled,"The contracting process was cancelled by the buyer or the procuring entity (for example, because of a change in needs, insufficient funds, or technical or procedural errors) after the award was made and the standstill period, if any, had ended."
```

odscjen commented on schema/codelists/contractFinalStatus.csv:
```diff
@@ -0,0 +1,5 @@
+terminatedSuccessfully,Terminated successfully,"The contract was concluded and in force, and is terminated due to its successful completion."
+terminatedEarly,Terminated early,"The contract was concluded and in force, and is terminated early due to some non-completion."
+cancelled,Cancelled,"The contracting process was cancelled by the buyer or the procuring entity (e.g. because of a change in needs, insufficient funds, or technical or procedural errors) after the award was made and the standstill period, if any, had ended."
+unsuccessful,Unsuccessful,"The contract failed (e.g. the winning bidder(s) refused to sign the contract) after the award was made and the standstill period, if any, had ended, but before the contract was concluded and in force."
```
```suggestion
unsuccessful,Unsuccessful,"The contract failed (for example, the winning bidder(s) refused to sign the contract) after the award was made and the standstill period, if any, had ended, but before the contract was concluded and in force."
```

odscjen commented on schema/dereferenced-release-schema.json:
```diff
@@ -6481,6 +6481,21 @@
+        },
+        "finalStatus": {
+          "title": "Tender final status",
+          "description": "The final status of the tender, from the closed [tenderFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#tender-final-status) codelist.",
```
```suggestion
          "description": "The final status of the tender, using the closed [tenderFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#tender-final-status) codelist.",
```
as per style guide https://ocds-standard-development-handbook.readthedocs.io/en/latest/meta/schema_style_guide.html#codelists

duncandewhurst replied:
Done in https://github.com/open-contracting/standard/pull/1648/commits/4bab921b7905faf32c1a52101b733deb20ea7eb5

odscjen commented on schema/dereferenced-release-schema.json:
```diff
@@ -9120,6 +9135,22 @@
+          },
+          "finalStatus": {
+            "title": "Award final status",
+            "description": "The final status of the award, from the closed [awardFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#award-final-status) codelist.",
```
```suggestion
            "description": "The final status of the award, using the closed [awardFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#award-final-status) codelist.",
```
as per https://ocds-standard-development-handbook.readthedocs.io/en/latest/meta/schema_style_guide.html#codelists

odscjen commented on schema/dereferenced-release-schema.json:
```diff
@@ -14588,6 +14619,23 @@
+          },
+          "finalStatus": {
+            "title": "Contract final status",
+            "description": "The final status of the contract, from the closed [contractFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#contract-final-status) codelist.",
```
```suggestion
            "description": "The final status of the contract, using the closed [contractFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#contract-final-status) codelist.",
```

odscjen commented on schema/dereferenced-release-schema.json:
```diff
@@ -20009,6 +20057,21 @@
+        },
+        "finalStatus": {
+          "title": "Tender final status",
+          "description": "The final status of the tender, from the closed [tenderFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#tender-final-status) codelist.",
```
```suggestion
          "description": "The final status of the tender, using the closed [tenderFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#tender-final-status) codelist.",
```

odscjen commented on schema/dereferenced-release-schema.json:
```diff
@@ -22644,6 +22707,22 @@
+        },
+        "finalStatus": {
+          "title": "Award final status",
+          "description": "The final status of the award, from the closed [awardFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#award-final-status) codelist.",
```
```suggestion
          "description": "The final status of the award, using the closed [awardFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#award-final-status) codelist.",
```

duncandewhurst replied:
Done in https://github.com/open-contracting/standard/pull/1648/commits/4bab921b7905faf32c1a52101b733deb20ea7eb5

odscjen commented on schema/dereferenced-release-schema.json:
```diff
@@ -28106,6 +28185,23 @@
+        },
+        "finalStatus": {
+          "title": "Contract final status",
+          "description": "The final status of the contract, from the closed [contractFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#contract-final-status) codelist.",
```
```suggestion
          "description": "The final status of the contract, using the closed [contractFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#contract-final-status) codelist.",
```

duncandewhurst replied:
Done in https://github.com/open-contracting/standard/pull/1648/commits/4bab921b7905faf32c1a52101b733deb20ea7eb5

odscjen commented on schema/release-schema.json:
```diff
@@ -517,6 +517,21 @@
+        },
+        "finalStatus": {
+          "title": "Tender final status",
+          "description": "The final status of the tender, from the closed [tenderFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#tender-final-status) codelist.",
```
```suggestion
          "description": "The final status of the tender, using the closed [tenderFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#tender-final-status) codelist.",
```

odscjen commented on schema/release-schema.json:
```diff
@@ -708,6 +723,22 @@
+        },
+        "finalStatus": {
+          "title": "Award final status",
+          "description": "The final status of the award, from the closed [awardFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#award-final-status) codelist.",
```
```suggestion
          "description": "The final status of the award, using the closed [awardFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#award-final-status) codelist.",
```

odscjen commented on schema/release-schema.json:
```diff
@@ -893,6 +924,23 @@
+        },
+        "finalStatus": {
+          "title": "Contract final status",
+          "description": "The final status of the contract, from the closed [contractFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#contract-final-status) codelist.",
```
```suggestion
          "description": "The final status of the contract, using the closed [contractFinalStatus](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#contract-final-status) codelist.",
```

duncandewhurst committed:
Merge branch '1.2-dev' into 1160-add-finalStatus-fields

duncandewhurst committed:
Update schema/codelists/awardFinalStatus.csv

Co-authored-by: odscjen <95221058+odscjen@users.noreply.github.com>

duncandewhurst committed:
Update schema/codelists/awardFinalStatus.csv

Co-authored-by: odscjen <95221058+odscjen@users.noreply.github.com>

duncandewhurst committed:
Update schema/codelists/contractFinalStatus.csv

Co-authored-by: odscjen <95221058+odscjen@users.noreply.github.com>

duncandewhurst committed:
Update schema/codelists/contractFinalStatus.csv

Co-authored-by: odscjen <95221058+odscjen@users.noreply.github.com>

duncandewhurst committed:
Update schema/release-schema.json

Co-authored-by: odscjen <95221058+odscjen@users.noreply.github.com>

duncandewhurst committed:
Update schema/release-schema.json

Co-authored-by: odscjen <95221058+odscjen@users.noreply.github.com>

duncandewhurst committed:
Update schema/release-schema.json

Co-authored-by: odscjen <95221058+odscjen@users.noreply.github.com>

duncandewhurst committed:
schema/dereferenced-release-schema.json: Run ./manage.py pre-commit

duncandewhurst committed:
Merge branch '1.2-dev' into 1160-add-finalStatus-fields

duncandewhurst commented:
Sorry about the long list of small commits. GitHub wouldn't let me commit the suggestions in a batch. Turns out you can't do that when there is a conflict with the base branch. I thought I'd resolved merge conflicts, but I hadn't realised another PR was merged into 1.2-dev since I merged 1.2-dev into this branch a few minutes ago!

@odscjen FYI - you don't need to review `dereferenced-release-schema.json` as that is automatically generated from the `release-schema.json` by the pre-commit script.

duncandewhurst commented:
@jpmckinney just checking that you saw the review request on this PR.

jpmckinney commented on docs/schema/codelists.md:
````diff
@@ -333,3 +333,33 @@ For enterprises without employees, use the 'micro' code.
 :file: ../../build/current_lang/codelists/partyScale.csv
 ```
+
+### Tender final status
````
All other headings are capitalized.

Putting these at the end is not best, but we'll be re-ordering in #1405 anyway.

duncandewhurst replied:
Done in 73513152975b11c70af6c233769996673ce1d1f7

jpmckinney commented on schema/release-schema.json:
```diff
@@ -517,6 +517,21 @@
           }
+        },
+        "finalStatus": {
+          "title": "Tender final status",
```
Same for the others.

```suggestion
          "title": "Final status",
```

duncandewhurst replied:
Done in 47c5ab09ffab6d5a09a0b0660f32460875cd10e6

jpmckinney commented on schema/codelists/tenderFinalStatus.csv:
```diff
@@ -0,0 +1,3 @@
+Code,Title,Description
+cancelled,Cancelled,"The contracting process was cancelled by the buyer or the procuring entity (for example, because of a change in needs, insufficient funds, or technical or procedural errors) after the contracting documents (for example, procurement documents) were available to potential supppliers, but not later than the bid submission deadline."
```
Missing 'complete'
```suggestion
complete,Complete,The bid submission deadline has passed.
cancelled,Cancelled,"The contracting process was cancelled by the buyer or the procuring entity (for example, because of a change in needs, insufficient funds, or technical or procedural errors) after the contracting documents (for example, procurement documents) were available to potential supppliers, but not later than the bid submission deadline."
```

jpmckinney reviewed (commented):
We need to rename statusDetails to finalStatusDetails, and re-order the fields so that details appears after status. (Use that same order for the two changelog entries.)

Codelists:

Adding more substantive comments here so that they don't become hard to find as resolved comments.

* terminatedSuccessfully was added to OCDS 1.2 in #1201 to close #395, and was named as such to be consistent with 'terminated', which was described as:

  > The contract was concluded and in force, and has now come to a close. This might be due to the successful completion of the contract, or might be early termination due to some non-completion.

  However, this new codelist can now be consistent with the other new codelists, and use 'complete' (both for the code and title) for the "happy" result.

* Re: terminatedEarly, to mirror 'complete', I suggest 'incomplete'.

Though it is often used in negative scenarios, I confirmed that "terminated" on its own can mean e.g. ["contract expiration"](https://www.spotdraft.com/blog/contract-termination-guide), so its use in the code descriptions is okay.

Examples:

* Update JSON and CSV examples to use `finalStatus` instead of `status` where possible, and to just remove `status` if not possible (e.g. if 'active'). Note: Milestone `status` remains correct.
* Update Markdown examples similarly. I think these are only: user_needs, change_history, unsuccessful_processes, framework_agreements.
* We need to revert #1344 re: contracts_suspension. I've updated the related issue #758.

codelists.md:

* Add `{deprecated}` directives to the old status codelists.
* Contract Status still has a `{versionchanged}` directive that should have been removed in #1509 when we removed those two codes.
* Remove the paragraphs for Tender Status, Award Status and Contract Status, as they add no new content compared to the field description (or in the case of Contract Status, needs to be removed viz contract_suspension).

Other changes:

* `releaseTag.csv` mentions status. I think just delete the last clause. The "pipeline" wording comes from #201 but it occurs nowhere else in OCDS.

duncandewhurst committed:
docs/schema/codelists.md: Capitalize headings

duncandewhurst committed:
release-schema: Update field titles

duncandewhurst committed:
Update schema/codelists/tenderFinalStatus.csv

Co-authored-by: James McKinney <26463+jpmckinney@users.noreply.github.com>

duncandewhurst committed:
release-schema: Rename statusDetails to finalStatusDetails

duncandewhurst committed:
schema/codelists/contractFinalStatus.csv: Rename codes

duncandewhurst committed:
docs/examples: Update status fields

duncandewhurst committed:
docs/guidance: Update references to status fields in Markdown examples

duncandewhurst committed:
examples: Remove contract_suspension

duncandewhurst committed:
docs/schema/codelists.md: Add deprecation directives, remove paragraphs

duncandewhurst committed:
schema/codelists/releaseTag.csv: Update planningUpdate description

duncandewhurst committed:
docs/examples: Reindent

duncandewhurst committed:
schema: Update codelist enums, run pre-commit script

duncandewhurst commented:
> We need to rename statusDetails to finalStatusDetails, and re-order the fields so that details appears after status. (Use that same order for the two changelog entries.)

Done in df3ca4eb6436fd3366f395209f74ca9fa3642246. `finalStatus` and `finalStatusDetails` appear at the end of each object. Let me know if you want them somewhere else. I combined the changelog entries since both changes are made in this PR.

> Codelists:
> 
> Adding more substantive comments here so that they don't become hard to find as resolved comments.
> 
>     * terminatedSuccessfully was added to OCDS 1.2 in [Add termination codes to contractStatus #1201](https://github.com/open-contracting/standard/pull/1201) to close [contractStatus: Add termination codes #395](https://github.com/open-contracting/standard/issues/395), and was named as such to be consistent with 'terminated', which was described as:
>       > The contract was concluded and in force, and has now come to a close. This might be due to the successful completion of the contract, or might be early termination due to some non-completion.
>       
>       
>       However, this new codelist can now be consistent with the other new codelists, and use 'complete' (both for the code and title) for the "happy" result.
> 
>     * Re: terminatedEarly, to mirror 'complete', I suggest 'incomplete'.
> 
> 
> Though it is often used in negative scenarios, I confirmed that "terminated" on its own can mean e.g. ["contract expiration"](https://www.spotdraft.com/blog/contract-termination-guide), so its use in the code descriptions is okay.

Done in 6ad1ff7814d86ecf1a15524ba0ef4b2258e9db06

> Examples:
> 
>     * Update JSON and CSV examples to use `finalStatus` instead of `status` where possible, and to just remove `status` if not possible (e.g. if 'active'). Note: Milestone `status` remains correct.

Done in 71f3226d3c11153c7dd618a316bdc0dd49eadacd and https://github.com/open-contracting/standard/pull/1648/commits/a3ef96f0dcca7b0b24955c529b932933104e6a8d.

>     * Update Markdown examples similarly. I think these are only: user_needs, change_history, unsuccessful_processes, framework_agreements.

Done in 9e32f1f1b1d1a4852013abc77a77d16724ef1b68

>     * We need to revert [Add contract suspension worked example #1344](https://github.com/open-contracting/standard/pull/1344) re: contracts_suspension. I've updated the related issue [Worked example (1.2): Contract suspension #758](https://github.com/open-contracting/standard/issues/758).

Done in 3fc8aa8dd5ce1c8268bc0750ee563056310e0a84

> codelists.md:
> 
>     * Add `{deprecated}` directives to the old status codelists.
> 
>     * Contract Status still has a `{versionchanged}` directive that should have been removed in [`tenderStatus`, `awardStatus` `contractStatus`: update codes #1509](https://github.com/open-contracting/standard/pull/1509) when we removed those two codes.
> 
>     * Remove the paragraphs for Tender Status, Award Status and Contract Status, as they add no new content compared to the field description (or in the case of Contract Status, needs to be removed viz contract_suspension).

Done in 1b9c1ca535417d3ad1df390b58b248eaf04a68a7

> Other changes:
> 
>     * `releaseTag.csv` mentions status. I think just delete the last clause. The "pipeline" wording comes from [Codelist updates: Introduce a 'planningUpdate' releaseTag and 'withdrawn' tenderStatus #201](https://github.com/open-contracting/standard/issues/201) but it occurs nowhere else in OCDS.

Done in 50598ca0b461248b411b9c85dcae81adbf1374da

duncandewhurst committed:
Merge branch '1.2-dev' into 1160-add-finalStatus-fields

duncandewhurst committed:
docs/examples: Reinstate tender.finalStatus = complete

duncandewhurst committed:
schema: Update enum, run pre-commit script

duncandewhurst committed:
schema/codelists/tenderStatus.csv: Update deprecation note for 'complete'

duncandewhurst commented:
@jpmckinney I think this PR is ready for review now. Two things to note:

1. In 0da5c89e0b428ca3fc0753afcca111d8ffd1a2c2, I updated the deprecation note for 'complete' in the tender status codelist to refer to 'complete' in the new tender final status codelist.
2. I noticed that `docs/examples/unsuccessful_tender/tender.json` has `lots.status` = 'unsuccessful' - should the lot extension also be updated following the same logic as the changes to `tender`?

jpmckinney commented:
> I noticed that docs/examples/unsuccessful_tender/tender.json has lots.status = 'unsuccessful' - should the lot extension also be updated following the same logic as the changes to tender?

Yes, please create an issue.

jpmckinney commented on docs/guidance/build/change_history.md:
```diff
@@ -64,9 +64,9 @@ The procuring entity makes the decision to award the contract to Balfour Beatty.
 The publisher creates a new OCDS release for the award. The record has the relevant information in the award section and uses the 'award' tag. 
 
-The `parties` array has a new entry with the supplier's information. The complete `tender` section is repeated, with the `status` field set to 'complete'. Previous releases remain unchanged.
+The `parties` array has a new entry with the supplier's information. The complete `tender` section is repeated. Previous releases remain unchanged.
```
Otherwise, the reference to `tender.finalStatus` in the next paragraph doesn't work.

```suggestion
The `parties` array has a new entry with the supplier's information. The complete `tender` section is repeated, and its `finalStatus` field is set to 'complete'. Previous releases remain unchanged.
```

jpmckinney commented on docs/guidance/build/change_history.md:
```diff
@@ -64,9 +64,9 @@ The procuring entity makes the decision to award the contract to Balfour Beatty.
+The `parties` array has a new entry with the supplier's information. The complete `tender` section is repeated. Previous releases remain unchanged.
 
-The publisher adds the new release to the record. They also update the compiled and versioned releases. The compiled release reflects the changes to the `awards`, `tender` and `parties` sections. The versioned release includes a new change for the `tender.status` field.
+The publisher adds the new release to the record. They also update the compiled and versioned releases. The compiled release reflects the changes to the `tender`, `awards` and `parties` sections. The versioned release includes a new change for the `tender.finalStatus` field.
```
```suggestion
The publisher adds the new release to the record. They also update the compiled and versioned releases. The compiled release reflects the changes to the `tender`, `awards` and `parties` sections. The versioned release now sets the `tender.finalStatus` field.
```

jpmckinney committed:
guidance/build/change_history: Normalize language for versioned release

jpmckinney committed:
guidance/map/unsuccessful_processes: Restore simple quotes

jpmckinney commented on docs/history/changelog.md:
```diff
@@ -311,6 +312,13 @@ Per the [normative and non-normative content and changes policy](../governance/n
 * [#1161](https://github.com/open-contracting/standard/pull/1161) Change recommendation for unknown time component.
 * [#1208](https://github.com/open-contracting/standard/pull/1208) Update guidance with new field definitions.
 * [#1216](https://github.com/open-contracting/standard/pull/1216) Update definitions of contracting process, record, and ocid. Introduce definition of planning process.
+* [#1307](https://github.com/open-contracting/standard/pull/1307) Clarify uniqueness rules for records.
+* [#1315](https://github.com/open-contracting/standard/pull/1315) Add rules on setting `id` and `date` for compiled releases to the merging specification.
+* [#1375](https://github.com/open-contracting/standard/pull/1375) Update guidance for empty fields in the merging documentation.
+* [#1466](https://github.com/open-contracting/standard/pull/1466) Reference worked examples in release and record reference documentation.
+* [#1466](https://github.com/open-contracting/standard/pull/1482) Add examples in release reference documentation.
+* [#1618](https://github.com/open-contracting/standard/pull/1618) Add conformance rule about normative statements.
+* [#1618](https://github.com/open-contracting/standard/pull/1618) Remove validator and application conformance rules.
```
These must be from a bad merge. They appear below in a reorganized fashion.
```suggestion
```

jpmckinney committed:
changelog: Remove contract suspension changelog entry 3fc8aa8

jpmckinney committed:
release-schema: Move finalStatus and finalStatusDetails after status

jpmckinney committed:
codelists: Remove versionchanged directive from deprecated contractStatus codelist

jpmckinney commented:
>> * Contract Status still has a `{versionchanged}` directive that should have been removed in [`tenderStatus`, `awardStatus` `contractStatus`: update codes #1509](https://github.com/open-contracting/standard/pull/1509) when we removed those two codes.
>
> Done in https://github.com/open-contracting/standard/commit/1b9c1ca535417d3ad1df390b58b248eaf04a68a7

I committed this now.

jpmckinney committed:
release-schema,tenderStatus(typo): Change favour to favor

jpmckinney committed:
tenderStatus: Use same Deprecated wording for 'complete' as for others

jpmckinney commented:
> In https://github.com/open-contracting/standard/commit/0da5c89e0b428ca3fc0753afcca111d8ffd1a2c2, I updated the deprecation note for 'complete' in the tender status codelist to refer to 'complete' in the new tender final status codelist.

I changed it to instead match the message for the other codes that got transferred to the new field.

jpmckinney committed:
guidance/build/change_history: Use note admonition for sidenote. Add link to example.

jpmckinney committed:
changelog: Remove second entry about tender.datePublished, which is new in 1.2

jpmckinney committed:
Apply suggestions from code review

guidance/build/change_history: Restore mention of finalStatus in merged releases. changelog: Remove repetitions.
